### PR TITLE
AirfRANS: gc=1.5 + WD=1e-2 + T_max=10 — compound regularization

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Grad-clip=1.5 is our new winning clip value (PR #2737: 0.00935 vs 0.01271 at gc=1.0, a 26.4% improvement). Weight decay 1e-2 previously compounded well with gc=1.0 (PR #2709: 0.01323 from 0.01419 at baseline clip). The hypothesis is that gc=1.5+WD=1e-2 compounds similarly — the stronger, more expressive gradient signal from clip=1.5 should pair with WD=1e-2 regularization for a further improvement beyond 0.00935.

This complements haku's PR #2743, which tests gc=1.5+T_max=5+WD=1e-2 (shorter cosine cycle). This PR uses the proven T_max=10 default to isolate the WD compound effect cleanly, then haku's PR will reveal if a faster cycle helps further.

Previous datapoints:
- gc=1.0, no WD: 0.01419 (PR #2680)
- gc=1.0, WD=1e-2: 0.01323 (PR #2709) — WD gives -6.8%
- gc=1.5, no WD: **0.00935** (PR #2737, current best)
- gc=1.5, WD=1e-2, T_max=10: **this experiment**

## Instructions

Run the following command exactly:

```bash
cd target/icml2026 && python train.py \
  --dataset airfrans \
  --airfrans_task full \
  --optimizer adamw \
  --lr 7e-4 \
  --cosine-t-max 10 \
  --grad-clip 1.5 \
  --weight-decay 1e-2 \
  --no-use-ema \
  --enable-fourier \
  --epochs 999 \
  --wandb_group airfrans-gc15-wd1e2
```

Changes vs. current best (PR #2737):
- Add `--weight-decay 1e-2` (WD regularization on top of gc=1.5)
- Everything else identical to the winning gc=1.5 config

No code changes needed — all flags are existing CLI args. Report `val_primary/surface_mse` at best epoch.

## Baseline

- **Primary metric:** `val_primary/surface_mse` (lower is better)
- **Current best:** 0.00935 (PR #2737, gc=1.5, T_max=10, AdamW lr=7e-4, no WD)
- **Previous WD compound (gc=1.0):** 0.01323 (PR #2709) → WD gave -6.8% at gc=1.0
- **External target:** 0.0043 (~2.2x gap remaining)
- **Reproduce current best:** `cd target/icml2026 && python train.py --dataset airfrans --airfrans_task full --optimizer adamw --lr 7e-4 --cosine-t-max 10 --grad-clip 1.5 --no-use-ema --enable-fourier --epochs 999`

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT fern:

## Results

**Verdict: NEGATIVE** — WD=1e-2 does NOT compound with gc=1.5. It significantly regresses the primary metric.

### Key Metrics

| Metric | This experiment (gc=1.5+WD=1e-2) | Baseline (gc=1.5, no WD) | Delta |
|---|---|---|---|
| **val_primary/surface_mse** | **0.02033** (epoch 34) | **0.00935** (epoch 40) | **+117% (2.17x worse)** |
| full_val/surface_mse_p | 0.0812 | — | — |
| full_val/volume_mse | 0.0899 | — | — |
| test_primary/surface_mse | 0.0204 | — | — |

### Trough envelope progression
- Epoch 14: 0.102 (phase transition onset — same timing as PR #2709)
- Epoch 19: 0.045 (first deep trough)
- Epoch 28: 0.033
- Epoch 34: **0.0203** (best)
- Epoch 38: 0.0221 (final deep trough — did NOT reach 0.01 range)

### Grad-clip stats
- 297 of 360 batches clipped at best epoch (82%)
- Mean grad norm: 16.9

### Run details
- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 7e-4 --cosine-t-max 10 --grad-clip 1.5 --weight-decay 1e-2 --no-use-ema --enable-fourier --epochs 999 --wandb-group airfrans-gc15-wd1e2 --agent fern --wandb-name "fern/airfrans-gc15-wd1e2-tmax10"`
- **W&B run:** h1m2zsvc
- **Epochs:** 41 (30-min timeout)
- **Best epoch:** 34

### What happened

WD=1e-2 compounded well with gc=1.0 (PR #2709: 0.01419 → 0.01323, -6.8%), but it **badly hurts** gc=1.5. The best surface_mse with WD was 0.0203, more than 2x worse than the gc=1.5 baseline of 0.00935.

The likely explanation: gc=1.5 already permits larger gradient updates than gc=1.0, enabling the model to explore deeper basins. Adding WD=1e-2 on top of that creates too much regularization — the weight penalty fights the larger gradient signal, preventing the model from reaching the very deep basins that gc=1.5 alone can find. The phase transition happened on schedule (epoch ~14), and the trough envelope descended steadily, but it plateaued around 0.02 instead of breaking through to 0.01.

Contrast with gc=1.0+WD: there, the tighter clipping meant gradients were already constrained, so WD's regularization effect dominated (smoothing the loss landscape). At gc=1.5, the additional gradient freedom IS the mechanism that produces the 0.00935 basin — WD undermines that freedom.

### Suggested follow-ups
- **WD=1e-3 at gc=1.5:** A lighter WD might provide some regularization benefit without overwhelming the gc=1.5 gradient signal
- **WD=1e-2 at gc=2.0:** If even looser clipping can compensate for WD, the compound might work at higher clip thresholds
- **gc=1.5 + longer training (no WD):** The baseline was still improving at epoch 40 — a 180-min run could push well below 0.00935

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2744 because ADVISOR: Closing — gc=1.5+WD=1e-2 achieved 0.020327, catastrophic dead end. WD fights gc=1.5's gradient headroom benefit. WD=1e-2 only works with gc=1.0 at 4L/256d..
